### ISSUE TITLE
Granule search based on its filename

### ISSUE THREAD

**Issue by slesaad** [nasa/eo-metadata-tools#18]
It would be useful for scientists to be able to search for a granule based on its filename. Currently, there is no way to do it.

**Comment by jceaser** [comment]
Are you talking about the GranuleMetaDataFile field?

**Comment by clynnes** [comment]
To clarify:  "As a scientist, I can paste the filename of a data file I am looking at into some TBD tool, and search CMR for the granule metadata record corresponding to that data file." (Assuming that I have not changed the filename from what it was when I originally downloaded it, natch.)

**Comment by jceaser** [comment]
that sounds like the echo onlineaccessurl AKA the related URL field

**Comment by clynnes** [comment]
Mmmm...I don't think so. We have only the filename, not the front part of the URL. Slesa and I were planning to loop through the providers until we found the filename in the readable_granule_id. (Seemed more efficient than looping through all 7500 collections until we found it.)

**Comment by clynnes** [comment]
(I do have a wild and crazy idea about inferring filename regex patterns, adding them as tags to the collection, and then searching through the collections until we find the regex matching the filename, then querying that collection.)

**Comment by jceaser** [comment]
I don't see any granule field which accepts wild cards, I do find ranges. Partial field search would require some discussion I think.

**Comment by clynnes** [comment]
No, that's not where the regex comes in. The regex is used to figure out which collection a data file came from. We can then search for the granule by its exact filename within a single collection.
